url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/TestForHttpRequestUri.java

```diff
@@ -0,0 +1,191 @@
+package net.hetimatan.util.http;
+
+import net.hetimatan.io.file.MarkableFileReader;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.util.http.HttpGetRequestUri;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import junit.framework.TestCase;
+
+public class TestForHttpRequestUri  extends TestCase {
+
+	public void testHello() {
+
+	}
+
+	public void testDecodeFragment() throws IOException {
+		{
+			MarkableFileReader reader = new MarkableFileReader("#nononono".getBytes());
+			assertEquals("#nononono", HttpGetRequestUri.fragment(reader));
+			reader.close();
+		}	
+	}
+
+	public void testDecodePath() throws IOException {
+		{
+			MarkableFileReader reader = new MarkableFileReader("/test/xxx.html".getBytes());
+			assertEquals("/test/xxx.html", HttpGetRequestUri.path(reader));
+			reader.close();
+		}
+
+	}
+	public void testDecodePort() throws IOException {
+		{
+			MarkableFileReader reader = new MarkableFileReader("8080".getBytes());
+			assertEquals(8080, HttpGetRequestUri.port(reader));
+			reader.close();
+		}
+
+
+		{
+			MarkableFileReader reader = new MarkableFileReader("808080808080808080808080".getBytes());
+			try {
+				HttpGetRequestUri.port(reader);
+				assertTrue(false);
+			} catch(NumberFormatException e) {
+				assertEquals(0, reader.getFilePointer());
+			} finally {
+				reader.close();
+			}
+
+		}
+	}
+
+	public void testDecodeHost() throws IOException {
+		{
+			MarkableFileReader reader = new MarkableFileReader("127.0.0.1".getBytes());
+			assertEquals("127.0.0.1", HttpGetRequestUri.host(reader));
+			reader.close();
+		}
+
+		{
+			MarkableFileReader reader = new MarkableFileReader("xxx.yyyy.xxx:".getBytes());
+			assertEquals("xxx.yyyy.xxx", HttpGetRequestUri.host(reader));
+			reader.close();
+		}
+
+		{
+			MarkableFileReader reader = new MarkableFileReader(":xxx.yyyy.xxx:".getBytes());
+			assertEquals("", HttpGetRequestUri.host(reader));
+			reader.close();
+		}
+	}
+
+	public void testDecodeScheme() throws IOException {
+		{
+			MarkableFileReader reader = new MarkableFileReader("http".getBytes());
+			assertEquals("http", HttpGetRequestUri.scheme(reader));
+			reader.close();
+		}
+
+		{
+			MarkableFileReader reader = new MarkableFileReader("https".getBytes());
+			assertEquals("https", HttpGetRequestUri.scheme(reader));
+			reader.close();
+		}
+
+		{
+			MarkableFileReader reader = new MarkableFileReader("htt".getBytes());
+			try {
+			assertEquals("htt", HttpGetRequestUri.scheme(reader));
+			assertTrue(false);
+			} catch(IOException e) {
+				
+			}
+			assertEquals(0, reader.getFilePointer());
+		}
+	}
+
+	public void testEncode001() throws IOException {
+		HttpGetRequestUri uri = new HttpGetRequestUri("*");
+		CashKyoroFile output = new CashKyoroFile(512);
+		uri.encode(output.getLastOutput());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getLastOutput*. Currently, there are execution paths that do not contain closure statements, for example, when HttpGetRequestUri.encode throws an exception. Ensure the object returned by *getLastOutput()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).